File: livingdoc-fixture-api/src/main/java/org/livingdoc/fixture/api/converter/time/LocalTimeConverter.java
package org.livingdoc.fixture.api.converter.time;

import java.time.LocalTime;
import java.time.format.DateTimeFormatter;


public class LocalTimeConverter extends AbstractTemporalConverter<LocalTime> {

    @Override
    DateTimeFormatter getDefaultFormatter() {
        return DateTimeFormatter.ISO_TIME;
    }

    @Override
    LocalTime parse(String value, DateTimeFormatter formatter) {
        return LocalTime.parse(value, formatter);
    }

}

File: livingdoc-fixture-converter/src/main/kotlin/org/livingdoc/fixture/converter/time/LocalTimeConverter.kt
package org.livingdoc.fixture.converter.time

import java.time.LocalTime
import java.time.format.DateTimeFormatter


open class LocalTimeConverter : AbstractTemporalConverter<LocalTime>() {

    override fun defaultFormatter(): DateTimeFormatter = DateTimeFormatter.ISO_LOCAL_TIME
    override fun doParse(value: String, formatter: DateTimeFormatter): LocalTime = LocalTime.parse(value, formatter)

    override fun canConvertTo(targetType: Class<*>) = LocalTime::class.java == targetType

}
